lib/nexthop_group.c

```diff
@@ -649,6 +649,30 @@ DEFPY(nexthop_group_backup, nexthop_group_backup_cmd,
 	return CMD_SUCCESS;
 }
 
+DEFPY(nexthop_group_allow_recursion,
+      nexthop_group_allow_recursion_cmd,
+      "[no] allow-recursion",
+      NO_STR
+      "Allow recursion for nexthops with no interface presence\n")
+{
+	VTY_DECLVAR_CONTEXT(nexthop_group_cmd, nhgc);
+
+	if (!!no == !CHECK_FLAG(nhgc->nhg.flags, NEXTHOP_GROUP_ALLOW_RECURSION))
```

Comment:
What does this check supposed to do?

Reply:
changing configuration or not.

Reply:
I'm not a huge fan of adding a new cli here.  Sharpd already refuses to install a nexhtop that doesn't have an interface attached to it.  Just make it auto turn on the flag if the nexhtop doesn't have an interface.

Reply:
In order to mimic BGP behaviour, I have to do this command. Or we limit sharpd for scalability testing, and in that case, the changes I do are not necessary from testing perspective with sharpd.